src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -87,9 +112,69 @@ public int getDimension() {
         return dimension;
     }
 
+    /**
+     * getter for model's state
+     *
+     * @return state
+     */
+    public ModelState getState() {
+        return state.get();
+    }
+
+    /**
+     * getter for model's timestamp
+     *
+     * @return timestamp
+     */
+    public TimeValue getTimestamp() {
+        return timestamp;
+    }
+
+    /**
+     * getter for model's description
+     *
+     * @return description
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * getter for model's error
+     *
+     * @return error
+     */
+    public String getError() {
+        return error;
+    }
+
+    /**
+     * setter for model's state
+     *
+     * @param state of the model
+     */
+    public synchronized void setState(ModelState state) {
+        this.state.set(Objects.requireNonNull(state, "state must not be null"));
+    }
+
+    /**
+     * setter for model's error
+     *
+     * @param error set on failure
+     */
+    public synchronized void setError(String error) {
+        this.error = Objects.requireNonNull(error, "error must not be null");
+    }
+
     @Override
     public String toString() {
-        return knnEngine.getName() + DELIMITER + spaceType.getValue() + DELIMITER + dimension;
+        return knnEngine.getName() + DELIMITER +
```

Comment:
nit: may be stringbuilder or String.join()

Reply:
Sure, will update.